### ISSUE TITLE
HTML5 + RDFa 1.1

### ISSUE THREAD

**Issue by chris-scheurle** [validator/validator#660]
[HTML+RDFa 1.1 - Second Edition](https://www.w3.org/TR/html-rdfa/#extensions-to-the-html5-syntax) states:

> For the avoidance of doubt, the following RDFa attributes are allowed on **all elements** in the HTML5 content model: `@vocab`, `@typeof`, `@property`, `@resource`, `@prefix`, `@content`, `@about`, `@rel`, `@rev`, `@datatype`, and `@inlist`. *[...]*
> * If the RDFa `@property` attribute is present on the `link` element, the `@rel` attribute is not required.
> * If the RDFa `@resource` attribute is present on the `link` element, the `@href` attribute is not required.
> * If the RDFa `@property` attribute is present on the `meta` element, neither the `@name`, `@http-equiv`, nor `@charset` attributes are required and the `@content` attribute **must** be specified.

But

```html
<link property="schema:isPartOf" resource="#website" typeof="schema:WebSite">
```

throws:

> Element `link` is missing required attribute `href`.

and 

```html
<link rel="preload" property="" href="resources/libraries/normalize.min.css" as="style">
```

throws:

> Attribute `as` not allowed on element `link` at this point.

**Comment by sideshowbarker** [comment]
```html
<link property="schema:isPartOf" resource="#website" typeof="schema:WebSite">
```
If that occurs as a descendant of the `body` element,  the checker will report no errors. But it’s intentionally not allowed in the `head`.

```html
<link rel="preload" property="" href="resources/libraries/normalize.min.css" as="style">
```
Omit the `property` attribute from that and it will validate.

**Comment by chris-scheurle** [comment]
@sideshowbarker wrote:

> If that occurs as a descendant of the `body` element, the checker will report no errors. But it’s intentionally not allowed in the `head`.

But that is in contradiction to the specs linked above. Might I ask what exactly that intention is?

@sideshowbarker wrote:

> Omit the `property` attribute from that and it will validate.

I know. But again, it *is* allowed by the specs. So the validator *should* accept it.
As for the reason why I'm doing this: some RDFa parsers tend to stumble on HTML5 link type values, an empty `property` attribute fixes that.

**Comment by chris-scheurle** [comment]
I just found another one:

```html
<dd property="telephone" content="+49XXXXXXXXXXX">0 XXX / XX XX XX XX</dd>
```

works fine in [validator.w3.org/nu/](http://validator.w3.org/nu/), but [html5.validator.nu/](http://html5.validator.nu/) throws:

> Attribute `content` not allowed on element `dd` at this point.

**Comment by sideshowbarker** [comment]
```html
<dd property="telephone" content="+49XXXXXXXXXXX">0 XXX / XX XX XX XX</dd>
```

> works fine in [validator.w3.org/nu/](http://validator.w3.org/nu/), but [html5.validator.nu/](http://html5.validator.nu/) throws

That’s because the `content` attribute is part of full RDFa but not part of RDFa Lite, and http://validator.w3.org/nu is the only instance that supports checking full RDFa — all other instances of the checker only support RDFa Lite

**Comment by sideshowbarker** [comment]
Pushed to https://validator.w3.org/nu/
